### PR TITLE
opentelemetry-cpp: add version 1.6.1 and support conan v2

### PR DESCRIPTION
**Pull request by toge**
Specify library name and version:  **opentelemetry-cpp/***

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by toge** [comment]
```
WARN: grpc/1.48.0: requirement abseil/20211102.0 overridden by opentelemetry-cpp/1.3.0 to abseil/20220623.0 
WARN: thrift/0.17.0: requirement boost/1.79.0 overridden by opentelemetry-cpp/1.3.0 to boost/1.80.0 
```
Need to upgrade dependencies of grpc and thrift.

**Comment by toge** [comment]
next missing dependencies is libevent.

**Comment by toge** [comment]
Thanks for https://github.com/conan-io/conan-center-index/pull/13520, missing dependencies for libevent is solved.

**Comment by prince-chrismc** [comment]
linters should be fixed to trigger it again

File: recipes/opentelemetry-cpp/all/conanfile.py [hunk 1]
```diff
@@ -24,47 +26,59 @@ class OpenTelemetryCppConan(ConanFile):
...
+        if self.info.settings.compiler.cppstd:
+            check_min_cppstd(self, self._minimum_cpp_standard)
+
+        if self.info.settings.arch != "x86_64":
+            raise ConanInvalidConfiguration(f"{self.ref} doesn't support architecture : {self.info.settings.arch}")
 
-        if (self.settings.compiler == "Visual Studio" and
-           tools.Version(self.settings.compiler.version) < "16"):
-            raise ConanInvalidConfiguration("Visual Studio 2019 or higher required")
+        if (self.info.settings.compiler == "Visual Studio" and Version(self.info.settings.compiler.version) < "16") or \
```
Comment (prince-chrismc) [inline recipes/opentelemetry-cpp/all/conanfile.py:74]: check min vs helper?

**Review by prince-chrismc — COMMENTED** [review]
one super tiny comment

File: recipes/opentelemetry-cpp/all/conanfile.py (hunk 1 above)
Comment (toge) [reply recipes/opentelemetry-cpp/all/conanfile.py]: Thank you for your review!
It is very useful helper.

Fixed.

**Review by prince-chrismc — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Pull request merged**